api/tests/identity-access-management/integration/infrastructure/repositories/user.repository.test.js

```diff
@@ -1183,6 +1186,74 @@ describe('Integration | Identity Access Management | Infrastructure | Repository
         expect(userDetailsForAdmin.isPixAgent).to.be.false;
       });
 
+      it('returns the found user when TOS feature toggle is true', async function () {
+        // given
+        sinon.stub(config, 'featureToggles').value({ isLegalDocumentsVersioningEnabled: true });
+        const createdAt = new Date('2021-01-01');
+        const emailConfirmedAt = new Date('2022-01-01');
+        const lastTermsOfServiceValidatedAt = new Date('2022-01-02');
+        const lastLoggedAt = new Date('2022-01-04');
+        const userInDB = databaseBuilder.factory.buildUser({
+          firstName: 'Henri',
+          lastName: 'Cochet',
+          email: 'henri-cochet@example.net',
+          cgu: true,
+          lang: 'en',
+          locale: 'en',
+          createdAt,
+          updatedAt: createdAt,
+          lastTermsOfServiceValidatedAt,
+          lastPixCertifTermsOfServiceValidatedAt: lastLoggedAt,
+          emailConfirmedAt,
+        });
+        await databaseBuilder.factory.buildUserLogin({ userId: userInDB.id, lastLoggedAt });
+        await databaseBuilder.commit();
+
+        const { PIX_ORGA } = LegalDocumentService.VALUES;
+        const { TOS } = LegalDocumentType.VALUES;
+
+        const service = PIX_ORGA;
+        const type = TOS;
+
+        const documentVersion = databaseBuilder.factory.buildLegalDocumentVersion({
+          service,
+          type,
+          versionAt: new Date('2024-02-01'),
+        });
+        const documentVersionUserAcceptance = databaseBuilder.factory.buildLegalDocumentVersionUserAcceptance({
+          userId: userInDB.id,
+          legalDocumentVersionId: documentVersion.id,
+          acceptedAt: new Date('2024-03-01'),
+        });
+        await databaseBuilder.commit();
+
+        // when
+        const userDetailsForAdmin = await userRepository.getUserDetailsForAdmin(userInDB.id);
+
+        // then
+        expect(userDetailsForAdmin).to.be.an.instanceOf(UserDetailsForAdmin);
```

Comment:
Suggestion: Avec un `domainBuilder`, contruire un `expectedUserDetailsForAdmin`. Remplacer les nombreux expects par un `expect(userDetailsForAdmin).to.deepEqualInstance(expectedUserDetailsForAdmin);`